oracle/oracle.go

```diff
@@ -366,3 +369,91 @@ func (o *oracleOps) Inspect(volumeIds []*string) ([]interface{}, error) {
 	}
 	return oracleVols, nil
 }
+
+// Create volume based on input template volume and also apply given labels.
+func (o *oracleOps) Create(template interface{}, labels map[string]string) (interface{}, error) {
+	vol, ok := template.(core.Volume)
+	if !ok {
+		return nil, cloudops.NewStorageError(cloudops.ErrVolInval,
+			"Invalid volume template given", "")
+	}
+
+	createVolReq := core.CreateVolumeRequest{
+		CreateVolumeDetails: core.CreateVolumeDetails{
+			CompartmentId:      &o.compartmentID,
+			AvailabilityDomain: &o.availabilityDomain,
+			SizeInGBs:          vol.SizeInGBs,
+			VpusPerGB:          vol.VpusPerGB,
+			DisplayName:        vol.DisplayName,
+			FreeformTags:       labels,
+		},
+	}
+	createVolResp, err := o.storage.CreateVolume(context.Background(), createVolReq)
+	if err != nil {
+		return nil, err
+	}
+
+	err = o.waitVolumeStatus(*createVolResp.Id, core.VolumeLifecycleStateAvailable)
+	if err != nil {
+		return nil, o.rollbackCreate(*createVolResp.Id, err)
+	}
+	return o.refreshVol(createVolResp.Id)
```

Comment:
refreshVol seems to be an extra unnecessary call. Make waitVolumeStatus return getVolResp and extract what you need there. That can be retuned. I am not sure why we have it for aws even.

Reply:
Done